### PR TITLE
Improve performance of address validation of interpreted code.

### PR DESCRIPTION
**Pull request by CristinaCristescu**

### CODE REVIEW

File: interpreter/cling/lib/Utils/Validation.cpp [hunk 1]
```diff
@@ -18,12 +19,41 @@
 # include <unistd.h>
 #endif
 
-
+#define CACHESIZE 8
```
Comment (CristinaCristescu) [inline interpreter/cling/lib/Utils/Validation.cpp]: I don't know if this is common practice in ROOT or not?

File: interpreter/cling/lib/Utils/Validation.cpp [hunk 2]
```diff
@@ -18,12 +19,41 @@
...
 namespace cling {
   namespace utils {
 
 #ifndef LLVM_ON_WIN32
+    struct Cache {
+    private:
+      const void* lines[CACHESIZE];
+      unsigned size, mostRecent;
+    public:
+      Cache(): lines{0,0,0,0,0,0,0,0}, size(CACHESIZE), mostRecent(0) {}
```
Comment (CristinaCristescu) [inline interpreter/cling/lib/Utils/Validation.cpp]: And I cannot use the CACHESIZE here..:-(

File: interpreter/cling/lib/Utils/Validation.cpp (hunk 1 above)
Comment (pcanal) [inline interpreter/cling/lib/Utils/Validation.cpp]: Please consider 'constexpr auto kCacheSize = 8;' and maybe define it in the class that needs it.

File: interpreter/cling/lib/Utils/Validation.cpp (hunk 2 above)
Comment (pcanal) [inline interpreter/cling/lib/Utils/Validation.cpp]: What do you mean?

File: interpreter/cling/lib/Utils/Validation.cpp (hunk 1 above)
Comment (karies) [inline interpreter/cling/lib/Utils/Validation.cpp]: I'm taking care of this - I'll discuss my changes with Cristina tomorrow. (E.g. here I suggest `std::array`.) Thanks nonetheless, Philippe!

**Comment by karies** [comment]
Thanks, merged (with a few refinements). Let's see what Jenkins says re performance...

**Pull request closed without merging**